test/jdk/java/awt/Multiscreen/MultiScreenCheckScreenIDTest.java

```diff
@@ -0,0 +1,148 @@
+    private static final int COLS = 12;
+    private static final int ROWS = 8;
+    private static final Color BACKGROUND = new Color(0, 0, 255, 64);
+    private static GraphicsDevice[] screens;
+    static List<Window> windowList = new ArrayList<>();
+    static Robot robot;
+    static JWindow window;
+
+
+    public static void main(final String[] args) throws Exception {
+        try {
+            createGUI();
+        } finally {
+            for (Window win : windowList) {
+                win.dispose();
+            }
+            if (window != null) {
+                window.dispose();
+            }
+        }
+        System.out.println("Test Pass");
+    }
+
+    private static void createGUI() throws AWTException {
+        new MultiScreenCheckScreenIDTest().createWindowGrid();
+    }
+
+    private void createWindowGrid() throws AWTException {
+        screens = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getScreenDevices();
+
+        if (screens.length < 2) {
+            System.out.println("Testing aborted. Required min of 2 screens. " +
+                    "Available : " + screens.length);
+            return;
+        }
+        robot = new Robot();
+
+        int screenNumber = 1;
+        for (GraphicsDevice screen : screens) {
+            Rectangle screenBounds = screen.getDefaultConfiguration().getBounds();
+
+            for (Rectangle r : gridOfRectangles(screenBounds, COLS, ROWS)) {
+                try {
+                    SwingUtilities.invokeAndWait(() -> {
+                        try {
+                            window = createWindow(r);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                } catch (InterruptedException | InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                robot.delay(50);
+                robot.waitForIdle();
+                if (window.getBounds().intersects(screenBounds)) {
+                    if (!(window.getGraphicsConfiguration().getBounds().
+                            intersects(screenBounds))) {
+                        throw new RuntimeException("Graphics configuration " +
+                                "changed for screen :" + screenNumber);
+                    }
+                }
+                windowList.add(window);
+            }
+            screenNumber++;
+        }
+    }
+
+    private JWindow createWindow(Rectangle bounds) {
+        JWindow window = new JWindow();
+        window.setBounds(bounds);
+        window.setBackground(BACKGROUND);
+        window.setAlwaysOnTop(true);
+        window.addMouseListener(this);
+        window.setVisible(true);
+        return window;
+    }
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+        ((Window) e.getSource()).dispose();
+    }
```

Comment:
In the automatic test, it's not necessary at all — you don't expect any input.

Reply:
Sometimes it's easier to run the test directly, without the jtreg.
Dispose is useful in this case to avoid locking the user's screen.

Reply:
It's *always* easier to run the test directly without jtreg.

It was in the original test case, you can dispose of a window by clicking. I see no real problem preserving it. :)